compiler/rustc_target/src/spec/apple_base.rs

```diff
@@ -63,15 +65,31 @@ fn deployment_target(var_name: &str) -> Option<(u32, u32)> {
         .and_then(|(a, b)| a.parse::<u32>().and_then(|a| b.parse::<u32>().map(|b| (a, b))).ok())
 }
 
-fn macos_deployment_target() -> (u32, u32) {
-    deployment_target("MACOSX_DEPLOYMENT_TARGET").unwrap_or((10, 7))
+fn macos_default_deployment_target(arch: &str) -> (u32, u32) {
+    if arch == "arm64" { (11, 0) } else { (10, 7) }
+}
+
+fn macos_deployment_target(arch: &str) -> (u32, u32) {
+    deployment_target("MACOSX_DEPLOYMENT_TARGET").unwrap_or(macos_default_deployment_target(arch))
```

Comment:
```suggestion
    deployment_target("MACOSX_DEPLOYMENT_TARGET").unwrap_or_else(|| macos_default_deployment_target(arch))
```
(Clippy warns on this, so someone is going to change this anyway sooner o later.)